packages/api/internal/orchestrator/cache.go

```diff
@@ -59,26 +58,33 @@ func (o *Orchestrator) keepInSync(ctx context.Context, instanceCache *instance.I
 			zap.L().Info("Stopping keepInSync")
 			return
 		case <-ticker.C:
-			o.syncNodes(ctx, instanceCache)
+			o.syncNodes(ctx, instanceCache, skipSyncingWithNomad)
 		}
 	}
 }
 
-func (o *Orchestrator) syncNodes(ctx context.Context, instanceCache *instance.InstanceCache) {
+func (o *Orchestrator) syncNodes(ctx context.Context, instanceCache *instance.InstanceCache, skipSyncingWithNomad bool) {
 	ctxTimeout, cancel := context.WithTimeout(ctx, cacheSyncTime)
 	defer cancel()
 
 	spanCtx, span := o.tracer.Start(ctxTimeout, "keep-in-sync")
 	defer span.End()
 
-	nomadNodes, err := o.listNomadNodes(spanCtx)
-	if err != nil {
-		zap.L().Error("Error listing orchestrator nodes", zap.Error(err))
-		return
-	}
-
 	var wg sync.WaitGroup
 
+	nomadNodes := make([]nodemanager.NomadServiceDiscovery, 0)
+
+	// Optionally, skip syncing from Nomad service discovery
+	if !skipSyncingWithNomad {
+		nomadSD, err := o.listNomadNodes(spanCtx)
+		if err != nil {
+			zap.L().Error("Error listing orchestrator nodes", zap.Error(err))
+			return
+		}
+
+		nomadNodes = nomadSD
+	}
+
```

Comment:
For tests we want to skip nomad service discovery. Starting with an empty list to make as few changes as needed.